manifests/charts/istio-cni/templates/daemonset.yaml

```diff
@@ -108,8 +110,10 @@ spec:
               # CAP_NET_RAW is required to allow iptables mutation of the `nat` table
               - NET_RAW
               # CAP_SYS_PTRACE is required for repair mode to describe the pod's network namespace
-              # in ambient and repair mode.
+              # in repair mode.
+{{- if .Values.repair.repairPods }}
```

Comment:
What exact thing does `repair` do that ambient does not with network namespaces that requires this?

They are separate features but they do nearly identical things with pod netnses, so I'm slightly surprised this is only required for one of them.

Reply:
When I was testing this change https://github.com/istio/istio/pull/53478 I got the following error when testing the repair controller: `failed to Statfs "/host/proc/1/ns/net": permission denied`. If the repair controller is not invoked or enabled I wasn't seeing this failure. I am digging into this more now.

Reply:
Ok - we [definitely `stat` procfs netns FDs in ambient too](https://github.com/istio/istio/blob/50dc03e621d9a457df479f8fe5f7b8241878d45d/cni/pkg/nodeagent/podcgroupns.go#L117), so I'd expect it to fail in both spots.

Reply:
Agree, working on writing tests now.

Reply:
I modified the repair controller values set [here](https://github.com/istio/istio/blob/master/tests/integration/ambient/main_test.go#L112) for the ambient integration tests to include `repairPods: false`. Based on the above changes this would remove CAP_SYS_PTRACE from the istio-cni-node's capabilities. With repair disabled, repair pods explicitly disabled, and the exclusion on CAP_SYS_PTRACE the ambient traffic tests still pass. I didn't expect this... looking into it

Reply:
Never mind, looks like there are some test failures. Will take a look.

Reply:
I am not able to reproduce the test failures locally. Adding ptrace cap back for ambient for now